files/dhcp/vrf

```diff
@@ -0,0 +1,178 @@
+
+                while [ $# -gt 0 ]; do
+                    net_length=$1
+                    via_arg=''
+
+                    case $net_length in
+                        32|31|30|29|28|27|26|25)
+                            if [ $# -lt 9 ]; then
+                                return 1
+                            fi
+                            net_address="${2}.${3}.${4}.${5}"
+                            gateway="${6}.${7}.${8}.${9}"
+                            shift 9
+                            ;;
+                        24|23|22|21|20|19|18|17)
+                            if [ $# -lt 8 ]; then
+                                return 1
+                            fi
+                            net_address="${2}.${3}.${4}.0"
+                            gateway="${5}.${6}.${7}.${8}"
+                            shift 8
+                            ;;
+                        16|15|14|13|12|11|10|9)
+                            if [ $# -lt 7 ]; then
+                                return 1
+                            fi
+                            net_address="${2}.${3}.0.0"
+                            gateway="${4}.${5}.${6}.${7}"
+                            shift 7
+                            ;;
+                        8|7|6|5|4|3|2|1)
+                            if [ $# -lt 6 ]; then
+                                return 1
+                            fi
+                            net_address="${2}.0.0.0"
+                            gateway="${3}.${4}.${5}.${6}"
+                            shift 6
+                            ;;
+                        0)  # default route
+                            if [ $# -lt 5 ]; then
+                                return 1
+                            fi
+                            net_address="0.0.0.0"
+                            gateway="${2}.${3}.${4}.${5}"
+                            shift 5
+                            ;;
+                        *)  # error
+                            return 1
+                            ;;
+                    esac
+
+                    # Take care of link-local routes.
+                    if [ "${gateway}" != '0.0.0.0' ]; then
+                        via_arg="via ${gateway}"
+                    fi
+
+                    # Set route (ip detects host routes automatically).
+                    ip -4 route add table ${table_id} "${net_address}/${net_length}" \
+                        ${via_arg} dev "${interface}" >/dev/null 2>&1
+                done
+            fi
+        fi
+
+        if [ -n "$alias_ip_address" ] &&
+           [ "$new_ip_address" != "$alias_ip_address" ]; then
+            ip -4 route add table ${table_id} ${alias_ip_address} dev ${interface} >/dev/null 2>&1
+        fi
+        ;;
+
+    EXPIRE|FAIL|RELEASE|STOP)
+        if [ -n "$alias_ip_address" ]; then
+            ip -4 route add table ${table_id} ${alias_ip_address} dev ${interface} >/dev/null 2>&1
+        fi
+
+        ;;
+
+    TIMEOUT)
```

Comment:
I do not under this part, why do we still add default route when the dhcp timeout? the rfc3442-classless-routes is much simpler. I thought here we only need to add the default route in the mgmt vrf so extending rfc3442 is much easier to review, unless the vrf script does more than what i describe above.

Reply:
The vrf script is more complete. Please see https://linux.die.net/man/8/dhclient-script. The rfc3442-classless-routes may be simpler but is not handling a lot of things.